### PR TITLE
address: consistent errors

### PR DESCRIPTION
**Pull request by dcousens**
This changes gives a more meaningful error for an invalid address,  while also making the exception message checking easier by making the exception format always contain the input value first (last would be fine too).

### CODE REVIEW

File: test/address.js [hunk 1]
```diff
@@ -20,17 +20,17 @@ describe('Address', function () {
...
+          Address.fromBase58Check(f.address)
+        }, new RegExp(f.address + ' ' + f.exception))
       })
     })
   })
 
   describe('fromOutputScript', function () {
     fixtures.valid.forEach(function (f) {
-      it('parses ' + f.script + ' (' + f.network + ')', function () {
+      it('parses ' + f.script.slice(0, 30) + '... (' + f.network + ')', function () {
```
Comment (dcousens) [inline test/address.js]: I also made the test descriptions shorter for readability

**Comment by dcousens** [comment]
@weilu please ACK post merge.

File: src/address.js [hunk 2]
```diff
@@ -4,9 +4,10 @@ var networks = require('./networks')
 var scripts = require('./scripts')
 var types = require('./types')
 
-function fromBase58Check (string) {
-  var payload = base58check.decode(string)
-  if (payload.length !== 21) throw new TypeError('Invalid address length')
+function fromBase58Check (address) {
+  var payload = base58check.decode(address)
+  if (payload.length < 21) throw new TypeError(address + ' is too short')
+  if (payload.length > 21) throw new TypeError(address + ' is too long')
```
Comment (dcousens) [inline src/address.js]: @weilu what are your thoughts on multiple exceptions in this case?  I know I've found it useful when dealing with the strict `DER` signature format.

**Comment by dcousens** [comment]
Fixed failing tests in https://github.com/bitcoinjs/bitcoinjs-lib/commit/eb752f83165e06d7dae2856f23c567758441cadd

File: src/address.js (hunk 2 above)
Comment (weilu) [inline src/address.js]: Nobody ever complains about the error message is too detailed. 

Nit: You can assign the error message based on the length check, and throw error based on the error message check.

**Comment by weilu** [comment]
GLTM

File: src/address.js (hunk 2 above)
Comment (dcousens) [inline src/address.js]: @weilu as in:

``` javascript
if (payload.length !== 21) {
  if (payload.length > 21) throw ...
  throw ...
}
```

Comment (weilu) [inline src/address.js]: I was talking about
```
var errorMessage;
if (payload.length < 21)  errorMessage = 'too short' 
else if (payload.length > 21)  errorMessage = 'too long' 

if (errorMessage !== undefined) throw new TypeError(errorMessage)
```

Meh, it was shorter in my mind :P Never mind then.

**Pull request merged**